<?xml version="1.0"?>
<launch>

  <!-- THIS BLOCK IS READY TO BE USED IN A NEW SIMULATION -->

  <!-- these are the arguments you can pass this launch file, for example paused:=true -->
  <!-- <arg name="model" default="$(find rt2_assignment1)/urdf/my_robot.urdf" /> -->

  <!-- We resume the logic in empty_world.launch 
   <include file="$(find gazebo_ros)/launch/empty_world.launch"> 
    <arg name="paused" value="false"/>
  </include> -->

  <!-- Load the URDF into the ROS Parameter Server
  <param name="robot_description" textfile="$(find rt2_assignment1)/urdf/my_robot.urdf"/> -->
	 
  <!-- Run a python script to the send a service call to gazebo_ros to spawn a URDF robot -->
  <!-- <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen"
	args="-urdf -model robot1 -param robot_description -z 0.05"/>  -->
	
  <!-- PARAMETERS FOR DUMMY NODES SIMULATION -->
  
     	<!-- Solution ID -->
   <arg name="solution_ID" default= "99999"/>	 <!-- dummy value -->
   
   	<!-- Room Numbers -->
   <arg name="room_number" default= "5"/>
  
  	<!-- Robot dummy pose -->
    <arg name="start_x" default= "0.0"/>
    <arg name="start_y" default= "0.0"/>
    <arg name="start_theta" default= "0.0"/>
    
    	<!-- Investigation time [s] -->
     <arg name="invest_time" default= "2.0"/>
     
    
  <!-- SET PARAMETERS SERVER VALUES -->     
    <param name="solution_ID" value="$(arg solution_ID)" />
    <param name="room_number" value="$(arg room_number)" />
    <param name="start_x" value="$(arg start_x)" />
    <param name="start_y" value="$(arg start_y)" />
    <param name="start_theta" value="$(arg start_theta)" />
    
    <param name="invest_time" value="$(arg invest_time)"/>
    
    
    
   <!-- Support nodes -->
   	<!-- Dummy camera -->
   <node name="dummy_image" pkg="cluedo_exp" type ="DummyCam.py"/>
        <!-- Dummy odometry -->
   <node name="dummy_odom" pkg="cluedo_exp" type ="dummy_odom.py"/> 


   <!-- Run the armor node -->
   <node pkg="armor" type="execute" name="armor_service" args="it.emarolab.armor.ARMORMainService"/> 
   
   <!-- Initialize the ontology -->
   <node name="armor_init" pkg="cluedo_exp" type ="armor_init.py" output="screen" /> 
   
   <!-- Dummy searche server -->
   <node name="ImageProcess_srv" pkg="cluedo_exp" type ="ImageProcessingUnit.py"  />    
   
   <!-- Get ID of Hypothesis server -->
   <node name="Get_ID_srv" pkg="cluedo_exp" type ="Get_Hyp_ID.py"  />  
   
   <!-- Consistency checker server -->
   <node name="Consistency_check_srv" pkg="cluedo_exp" type ="consistency_check.py"  output="screen" />    
   
   <!-- Oracle query -->
   <node name="Oracle_Query_srv" pkg="cluedo_exp" type ="Oracle_Query.py" output="screen" />   
   
   <!-- Move_to action server --> 
   <node name="move_to" pkg="cluedo_exp" type ="move_to.py" output="screen" />  
	
   <!-- Controller -->	
   <node name="controller" pkg="cluedo_exp" type ="controller.py"  output="screen" />   


      
   <!-- <node name="userInterface" pkg="rt2_assignment1" type="user_interface.py" output="screen"/> -->
   

</launch>
